sydent/threepid/bind.py

```diff
@@ -125,6 +125,22 @@ def _notify(self, assoc, attempt):
         else:
             logger.info("Successfully notified on bind for %s" % (mxid,))
 
+            # Only remove sent tokens when they've been successfully sent.
+            if assoc.get("invites"):
+                try:
+                    joinTokenStore = JoinTokenStore(self.sydent)
+                    joinTokenStore.deleteTokens(assoc["medium"], assoc["address"])
+                    logger.info(
+                        "Successfully deleted invite for %s from the store",
+                        assoc["address"],
+                    )
+                except Exception as e:
+                    logger.error(
```

Comment:
I'd suggest `logger.exception` rather than `logger.error(msg, e)`, so that we get the stack trace.